### ISSUE TITLE
All help files are generated empty

### ISSUE THREAD

**Issue by ChaseFlorell** [nightroman/Helps#6]
Hey there. I'm trying to iterate over all of the functions in my module, and generate docs for them. The issue I'm running into is that the files don't contain anything (namely the synopsis).

This is how I'm running it

    . "$($this.packagesDir)\Helps.1.1.4\tools\Helps.ps1"
    Get-Command -Module 'poshBAR' | % {
        New-Helps -command $_.Name > "$($this.documentationDir)\$_-Help.ps1"
        Convert-Helps "$($this.documentationDir)\$_-Help.ps1" "$($this.documentationDir)\$_.xml"
    }

Here's an example of one of the functions.

    <#
        .DESCRIPTION
            Adds the specified line to the hosts file if it doesn't already exist.
    
        .EXAMPLE
            Add-HostsFileEntry "local.example.ca"
    
        .EXAMPLE
            Add-HostsFileEntry "local.example.ca" "10.1.1.1"
    
        .EXAMPLE
            Add-HostsFileEntry "local.example.ca" -includeLoopbackFix
    
        .PARAMETER ipAddress
            The ip address of the machine you want to target.
    
        .PARAMETER hostName
            The "host name" name you want routed to the target ip address.
    
        .PARAMETER includeLoopbackFix
            Switch to determine if the Loopback Fix should also be applied (Add-LoopbackFix)
    
        .SYNOPSIS
            Will add a hosts file entry for the host name specified targeting the specified ip address.
    
        .NOTES
            Nothing yet...
    #>
    function Add-HostsFileEntry
    {
        [CmdletBinding()]
    	param(
            [parameter(Mandatory=$true,position=0)] [string] $hostName,
            [parameter(Mandatory=$false,position=1)] [ValidatePattern('\d{1,3}(\.\d{1,3}){3}')] [string] $ipAddress = "127.0.0.1",
            [parameter(Mandatory=$false,position=2)][switch] $includeLoopbackFix
    	)
    
        $ErrorActionPreference = "Stop"
        Write-Host ($msgs.msg_add_host_entry -f $hostName) -NoNewLine
    
    	$HostsLocation = "$env:windir\System32\drivers\etc\hosts"
    	$NewHostEntry = "`t$ipAddress`t$hostName"
    
    
    	if((gc $HostsLocation) -contains $NewHostEntry)
    	{
            Write-Host "`tExists" -f Cyan
    	}
    	else
    	{
            Write-Host "`tDone"
    		Add-Content -Path $HostsLocation -Value $NewHostEntry
    	}
    
    	# Validate entry
        Write-Host ($msgs.msg_validate_host_entry -f $hostName) -NoNewLine
    	if((gc $HostsLocation) -contains $NewHostEntry)
    	{
            Write-Host "`tPassed" -f Green
    	}
    	else
    	{
            Write-Host "`tFailed" -f Red
    	}
    
        if($includeLoopbackFix.IsPresent){
            Add-LoopbackFix $hostName
        }
    }

And here's the output it's giving me.

    # Add-HostsFileEntry command help
    @{
    	command = 'Add-HostsFileEntry'
    	synopsis = ''
    	description = ''
    	parameters = @{
    		hostName = ''
    		includeLoopbackFix = ''
    		ipAddress = ''
    	}
    	inputs = @(
    		@{
    			type = ''
    			description = ''
    		}
    	)
    	outputs = @(
    		@{
    			type = ''
    			description = ''
    		}
    	)
    	notes = ''
    	examples = @(
    		@{
    			#title = ''
    			#introduction = ''
    			code = {
    			}
    			remarks = ''
    			test = { . $args[0] }
    		}
    	)
    	links = @(
    		@{ text = ''; URI = '' }
    	)
    }

Have you any clue what I might be doing wrong? Am I right in assuming that the help files should be generated off of the `<# #>` comments at the top of the method?

**Comment by nightroman** [comment]
If you already have written help in comments then you do not have to generate
anything. Just call `Get-Help` and help will be shown (if it is correct). In other words,
you do not need `Helps`.

`Helps` is designed for a different scenario. You do not write help in
comments. Instead for every command you call `New-Helps` once to generate empty
templates. You write your help in these generated templates. When your functions
change you update help script accordingly. `Convert-Helps` will tell you that
some new parameters are not documented or some parameters are invalid (e.g. you
renamed or removed some). The last piece: in your functions you add help comment
links to the generated XML:

    #.ExternalHelp Helps-Help.xml
    function Convert-Helps(

Take a look at [Helps.ps1](https://github.com/nightroman/Helps/blob/master/Helps.ps1).
Its functions do not have help comments except links.

In other words, if you still want to use `Helps` (because it has advantages)
you have to move your existing help comments to the generated template help
scripts manually and replace help comments with help links to XML produced from
help scripts by `Convert-Helps`.
